### PR TITLE
Focused Launch: Pixel perfect loading state

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

* This PR changes the domain picker a bit to show an accurate loading state in the focused launch. 

I realize it's a backlog issue, but the loading was terribly janky that it might make the upcoming CfT look bad.

**Before:**
https://cloudup.com/c5uPXDg7x6l

**After:**
loading state

#### Testing instructions

1. In apps/editing-toolkit run `yarn dev --sync`.
2. Sandbox YOUR_SITE.wordpress.com.
3. Visit https://YOUR_SITE.wordpress.com/wp-admin/post-new.php.
4. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.
5. You should see loading similar to the one in the GIF above.

Fixes https://github.com/Automattic/wp-calypso/issues/47410

### CODE REVIEW

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@alshakero please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/953396). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: packages/launch/src/hooks/use-site.ts
Comment (razvanpapadopol) [inline packages/launch/src/hooks/use-site.ts:19]: Is there a reason for not using `isFetchingSite` selector from `SITE_STORE`? This applies also to https://github.com/Automattic/wp-calypso/pull/47528/files#diff-8de21fd6530118850c1d099df9447b90bf23dd1fce2871d58da41634b06f0c7dR21

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:104]: Should be grouped with the `hasPaidDomain` prop

**Review by ciampo — COMMENTED** [review]
When the screen is narrower than large desktop, the UI still jumps when the data loads

launch-loading-jump

File: packages/domain-picker/src/domain-picker/index.tsx
Comment (ciampo) [inline packages/domain-picker/src/domain-picker/index.tsx:324]: Although I understand what this code is doing, it may not be very clear in the future / to a developer with little context.

We should either refactor the code to make it a bit more explicit or at least add a comment explaining that, while the dependencies are loading, we show a specific number of placeholder items.

**Comment by ciampo** [comment]
Also, just a heads up — we should rebase this PR (or merge master into it) before approving and merging, as the changes to `useDomainSearch()` hook may break the code from #47318

File: packages/domain-picker/src/domain-picker/index.tsx
Comment (alshakero) [reply packages/domain-picker/src/domain-picker/index.tsx:324]: Great point.

File: packages/launch/src/hooks/use-site.ts
Comment (alshakero) [reply packages/launch/src/hooks/use-site.ts]: Didn't know about this one!

Comment (alshakero) [reply packages/launch/src/hooks/use-site.ts:19]: I tried it and it's never `true` 🤷

**Comment by alshakero** [comment]
> When the screen is narrower than large desktop, the UI still jumps when the data loads

I can't think of an easy fix for this one because we can't predict the height. The easiest would be to add a transition to height, which is still too complicated to implement.

**Comment by alshakero** [comment]
Addressed all. Thanks a lot!

**Comment by ciampo** [comment]
>I can't think of an easy fix for this one because we can't predict the height. The easiest would be to add a transition to height, which is still too complicated to implement.

Fair enough. I just felt like reporting it for completeness

**Review by ciampo — COMMENTED** [review]
LGTM, but I'll defer to @razvanpapadopol for the final approval after he's happy with  https://github.com/Automattic/wp-calypso/pull/47528#discussion_r526008372

File: packages/domain-picker/src/domain-picker/index.tsx
Comment (razvanpapadopol) [inline packages/domain-picker/src/domain-picker/index.tsx:78]: From the perspective of this component, what do you think about naming this prop `forceLoadingState`?

File: packages/launch/src/hooks/use-domain-search.ts
Comment (razvanpapadopol) [inline packages/launch/src/hooks/use-domain-search.ts:26]: Both are always equal since they are reflecting the loading state of the site. Proposed a simplification in https://github.com/Automattic/wp-calypso/pull/47554/files#diff-fd3904ca53e493ed641e3deffd5efb5fd5a320c2aa489342a0c820f6dc573044R26

File: packages/launch/src/hooks/use-site.ts
Comment (razvanpapadopol) [reply packages/launch/src/hooks/use-site.ts:19]: You're right. Added the correct reducer in https://github.com/Automattic/wp-calypso/pull/47554/files#diff-df112a4c704268086d2d3100bede88722f9193c88c107359dc4c401ccf97834fR60

**Review by razvanpapadopol — APPROVED** [review]
Thanks for addressing this issue and making the initial user experience much smoother. I've added a **Before** demo in PR description for comparison.

I've proposed some light improvements in https://github.com/Automattic/wp-calypso/pull/47554 and suggested a prop name update but there is clearly nothing blocking.

**Pull request merged**
